wcomponents-core/src/test/java/com/github/bordertech/wcomponents/WebUtilities_Test.java

```diff
@@ -517,6 +518,85 @@ public void testGetParentNamingContext()
     }
+    
+    /**
+     * Test the function of updateBeanValue() for all variants (single argument, ignoreVisble [true|false]).
+     */
+    @Test
+    public void testUpdateBeanValue()
+    {
+        runUpdateBeanValue(null);
+        runUpdateBeanValue(true);
+        runUpdateBeanValue(false);
+    }
+    
+    /**
+     * Set up and execute the updateBeanValue method with the given parameter.
+     * If the parameter is null then the default updateBeanValue(component) method will be invoked.
+     * 
+     * @param ignoreVisible the parameter to pass to WebUtilities.updateBeanValue(component, ignoreVisible).
+     */
+    private void runUpdateBeanValue(final Boolean ignoreVisible)
+    {
+        final String directChild = "directChild";
+        final String grandChild = "grandChild";
+        final String invisibleGrandChild = "invisibleGrandChild";
+        final String childOfInvisibleContainer = "childOfInvisibleContainer";
+        
+        Map<String, String> beanMap = new HashMap<>();
+        beanMap.put(directChild, null);
+        beanMap.put(grandChild, null);
+        beanMap.put(invisibleGrandChild, null);
+        beanMap.put(childOfInvisibleContainer, null);
+        
+        WContainer root = new WContainer();
+        root.setBean(beanMap);
+        WTextField childTextField = new WTextField();
+        childTextField.setBeanProperty(directChild);
+        childTextField.setData(directChild);
+        childTextField.setFlag(ComponentModel.USER_DATA_SET, true);
+        root.add(childTextField);
+        
+        WContainer childContainer = new WContainer();
+        root.add(childContainer);
+        WTextField grandChildTextField = new WTextField();
+        grandChildTextField.setBeanProperty(grandChild);
+        grandChildTextField.setData(grandChild);
+        grandChildTextField.setFlag(ComponentModel.USER_DATA_SET, true);
+        childContainer.add(grandChildTextField);
+        
+        WTextField invisibleGrandChildTextField = new WTextField();
+        invisibleGrandChildTextField.setBeanProperty(invisibleGrandChild);
+        invisibleGrandChildTextField.setData(invisibleGrandChild);
+        invisibleGrandChildTextField.setVisible(false);
+        invisibleGrandChildTextField.setFlag(ComponentModel.USER_DATA_SET, true);
+        childContainer.add(invisibleGrandChildTextField);
+        
+        
+        WContainer invisibleContainer = new WContainer();
+        invisibleContainer.setVisible(false);
+        root.add(invisibleContainer);
+        WTextField childOfInivisbleContainerTextField = new WTextField();
+        childOfInivisbleContainerTextField.setBeanProperty(childOfInvisibleContainer);
+        childOfInivisbleContainerTextField.setData(childOfInvisibleContainer);
+        childOfInivisbleContainerTextField.setFlag(ComponentModel.USER_DATA_SET, true);
+        invisibleContainer.add(childOfInivisbleContainerTextField);
+        

```

Comment:
To save calling the USER_DATA_SET flag (as you should not have to call this), you need to "lock" the component tree so it treats the test like a "new user" session. When the tree is locked this flag will be set. It is not being set at the moment as WComponents is treating any values "Set" as the "defaults".

So after all the components have been setup, you can include this....

root.setLocked(true);
setActiveContext(createUIContext());